colossalai/shardformer/layer/attn.py

```diff
@@ -444,18 +445,11 @@ def get_double_ring_groups(sp_group, inner_ring_size=None):
         sp_size = dist.get_world_size(sp_group)
         sp_rank = dist.get_rank(sp_group)
 
-        if inner_ring_size is None:
-            if torch.cuda.device_count() >= dist.get_world_size():
-                # single node, no need to consider NICs
-                return sp_group, sp_group
-            if sp_size <= 4:
-                inner_ring_size = min(2, sp_size)
-            else:
-                inner_ring_size = min(4, sp_size)
```

Comment:
We can keep this rule of thumb logic for easy topology construction? Users may have no idea what double ring means and not touch `inner_ring_size` at all

Reply:
@ver217 said it's better to keep it flexible as we might meet various kinds of machines. The users can use double ring by setting inner_ring_size to a non-none value.